### PR TITLE
Rewrite generator

### PR DESCRIPTION
**Pull request by jackieokay**
TODO:

- [x] Fix Unicode bug
- [x] Compile tests

### CODE REVIEW

File: src/svd_parser/format_utils.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    return [ports]
+
+
+def port_number(port):
+    if not port.isdigit():
+        return ord(port) - ord('A')
+    return port
+
+
+# Technically this is a generic property function and should go in parser_utils
```
Comment (CvRXX) [inline src/svd_parser/format_utils.py]: I think you mean format_utils?

**Review by CvRXX — COMMENTED** [review]
Looks good in general, hope to find time this week to test with actual hardware...

**Comment by jackieokay** [comment]
Ping @CvRXX, I fixed a few subtle bugs and I think this is ready for testing on hardware.

**Comment by jackieokay** [comment]
To generate files:

mkdir build
cd build
cmake .. && make

this expects you to have a folder called "data" containing CMSIS SVD files. I haven't committed this to the repo yet since I'm undecided if it should be copied in or if there should be a build step that clones the SVD files from another repo.

**Comment by CvRXX** [comment]
I'll try to test before fosdem but I also have to move this week to a new apartment so we'll see how it goes... 

On 30 January 2017 15:50:21 CET, Jackie Kay <notifications@github.com> wrote:
>Ping @CvRXX, I fixed a few subtle bugs and I think this is ready for
>testing on hardware.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/kvasir-io/HardwareDescriptionFileGenerator/pull/6#issuecomment-276081523

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Pull request merged**